### PR TITLE
Fixes #2379 Import coverage file error

### LINKED ISSUE #2379: Import coverage file error
Create new python application project
Paste the following code in a `program.py` file:
```
import time

def f():
        for i in range(10000):
                time.sleep(0)

f()
```

From command line, run the profiler:
```
py -3.6 -m coverage run program.py
py -3.6 -m coverage xml
```

Close `program.py` if it's open in  the editor.

Tools, Python Tools, Import coverage file
Select coverage.xml from the project folder

Result:
```
---------------------------
Microsoft Visual Studio
---------------------------
Value does not fall within the expected range.
---------------------------
OK   
---------------------------
```

It's thrown by this piece of code, which is trying to get to the project, with the goal of getting the python environment.  When the filename is not open in the editor, `IVsRunningDocumentTable4.GetDocumentCookie` throws instead of returning `VSCOOKIE_NIL`.

```
        internal static PythonProjectNode GetProjectFromFile(this IServiceProvider serviceProvider, string filename) {
            var docTable = serviceProvider.GetService(typeof(SVsRunningDocumentTable)) as IVsRunningDocumentTable4;
            var cookie = docTable.GetDocumentCookie(filename);
            if (cookie != VSConstants.VSCOOKIE_NIL) {
```

### PR DESCRIPTION
**Pull request by zooba**
Fixes #2379 Import coverage file error
Switches AnalysisEntryService to use existing helper function
Minor refactoring

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Extensions.cs [hunk 1]
```diff
@@ -375,22 +375,27 @@ public static class Extensions {
 
         internal static PythonProjectNode GetProjectFromFile(this IServiceProvider serviceProvider, string filename) {
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/Extensions.cs]: I'd like to make it clear that this is not going to return the project if the file isn't opened in an editor.  Maybe add a summary comment or rename the method to make it more obvious.

**Comment by huguesv** [comment]
The version used for the parser may end up being wrong just because none of the files were opened in the editor and the global default was picked, but I suppose that's not so terrible.

👍

**Comment by zooba** [comment]
You're right - we should be searching all open projects here as well before falling back. Let me revise the API a bit.

**Comment by zooba** [comment]
@huguesv New code added - can you take another look?

**Comment by huguesv** [comment]
Looks ok 👍

**Pull request merged**